### ISSUE TITLE
Extension causes Gnome to "crash" when switching on Dark theme

### ISSUE THREAD

**Issue by LeXofLeviafan** [mukul29/legacy-theme-auto-switcher-gnome-extension#4]
I haven't been able to test out the extension properly, as it doesn't seem to do anything if I enable it with Dark theme on; if, on the other hand, I turn it on with the Light theme (or switch to it without disabling the extension) and then switch back to Dark, I get a full-screen modal saying that "something went wrong, you must log out" with a single "Log out" button. After I log in back into the session, the extension still doesn't seem to have any effect.

Shell version: 42 (43 isn't stable yet in my distro)

**Comment by mukul29** [comment]
Not sure what the issue is here as I've not experienced something like this yet. Were all other extensions disabled when you tried this?

**Comment by LeXofLeviafan** [comment]
Yes, the issue happens even when all other extensions are disabled.

Switching to Light causes the desktop to flicker, and sends me back to Activities view, but doesn't cause any further problems. Switching to Dark, however, forces me to logout.

**Comment by mukul29** [comment]
Are you getting any logs from gnome-shell?

**Comment by LeXofLeviafan** [comment]
Testing that isn't exactly a trivial matter, considering I basically need to stop all my current work in all applications to prepare for the upcoming crash…

After testing it with system log open in framebuffer terminal, I can see that when the desktop "flickers" (which happens regardless of which theme is being switched on), a bunch of errors is printed to the system log, e.g.:
```
gsd-<daemon>[<pid>]: segfault at <hexcode> ip 0000<same-hexcode> sp 0000<other-hexcode> error 14 in <library, "user" or "mime.cache">[<segment>]
Code: Unable to access opcode bytes at RIP 0x<another-hexcode>.
```
And in case of a "crash" (which this time happened when turning on Light theme for the 2nd time, smh) there's an additional line between `gsd-power` and `gsd-xsettings`:
```
elogind-daemon[<pid>]: Removed session <number>.
```
